File: cmap/models/map_spine.py
"""
MapSpine is a subclass Quad and QuadView
"""

from os import path
from bokeh.models.glyphs import Quad
from bokeh.util.compiler import FromFile

script_path = path.dirname(path.abspath( __file__ ))

class MapSpine(Quad):

    __implementation__ = FromFile(script_path + '/map_spine.js')

    def __init__(self, **kwargs):
        Quad.__init__(self, **kwargs)

File: cmap/models/map_spine.js
import {Quad, QuadView} from 'models/glyphs/quad';

class MapSpineView extends QuadView {
}


export class MapSpine extends Quad {
}
MapSpine.prototype.default_view = MapSpineView;
MapSpine.prototype.type = 'MapSpine'; // name of the Python class
